### PR TITLE
Stabilize openstack_test when running against real cloud

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

in TestReadConfig, we are setting some env vars for testing if
we read them back properly. However this interferes with running
the unit test harness against a real openstack cloud where we
source the OS_* environment variables. Adding code here to save
and reset variables.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/openstack/openstack_test.go [hunk 1]
```diff
@@ -88,6 +88,11 @@ func TestReadConfig(t *testing.T) {
 		t.Errorf("Should fail when no config is provided: %s", err)
 	}
 
+	// Since we are setting env vars, we need to reset old
+	// values them for other tests to succeed.
```
Comment (dixudx) [inline pkg/cloudprovider/providers/openstack/openstack_test.go]: them? A typo here.

Comment (dims) [reply pkg/cloudprovider/providers/openstack/openstack_test.go]: oops. fixing

**Comment by dims** [comment]
@dixudx this is ready now. PTAL.

**Pull request merged**